elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+                'label': label,
+                'description': description,
+                'image': "",
+                'x_pos': 0,
+                'y_pos': 0
+            }
+        }
+    }
+
+    return node_type
+
+
+class ComponentParser(SingletonConfigurable):
+    _type = "local"
+    properties: dict() = {}
+    parameters: dict() = {}
+
+    def __init__(self):
+        super().__init__()
+        self.properties = self.get_common_config('properties')
+
+    def get_common_config(self, config_name):
+        common_dir = os.path.join(os.path.dirname(__file__), 'resources')
+        common_file = os.path.join(common_dir, f"{config_name}.json")
+        with io.open(common_file, 'r', encoding='utf-8') as f:
+            common_json = json.load(f)
+
+        return common_json
+
+    def _get_component_catalog_json(self):
+        catalog_dir = os.path.join(os.path.dirname(__file__), "resources")
+        catalog_file = os.path.join(catalog_dir, f"{self._type}_component_catalog.json")
+        with open(catalog_file, 'r') as f:
+            catalog_json = json.load(f)
+
+        return catalog_json['components']
+
+    def list_all_components(self):
+        components = []
+        if self._type != "local":
+            for component in self._get_component_catalog_json():
+                assert "path" in component
+                components.append(component)
+
+        return components
+
+    def return_component_if_exists(self, component_id):
+        for component in self._get_component_catalog_json():
+            if component['id'] == component_id:
+                return component
+        return None
+
+    def add_component(self, request_body):
+        component_json = {
+            'name': request_body['name'],
+            'id': get_id_from_name(request_body['name']),
+            'path': request_body['path']
+        }
+
+        catalog_json = self._get_component_catalog_json()
+        catalog_json['components'].append(component_json)
+
+        catalog_dir = os.path.join(os.path.dirname(__file__), "resources")
+        catalog_file = os.path.join(catalog_dir, f"{self._type}_component_catalog.json")
+        with open(catalog_file, 'a') as f:
+            f.write(json.dumps(catalog_json))
+
+    def parse_component_details(self, component, component_name):
+        """Get component name, id, description for palette JSON"""
+        pass
+
+    def parse_component_properties(self, component):
+        """Get component properties for properties JSON"""
+        raise NotImplementedError
+
+
+class KfpComponentParser(ComponentParser):
+    _type = "kfp"
+
+    def __init__(self):
+        super().__init__()
+
+    def parse_component_details(self, component_body, component_name=None):
+        component_json = {
+            'label': component_body['name'],
+            'image': "",
+            'id': get_id_from_name(component_body['name']),
+            'description': ' '.join(component_body['description'].split()),
```

Comment:
Is `component_body['description']` a required value?  If not, this will result in an exception.

Reply:
Great point. Should definitely do some checking around this